sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java

```diff
@@ -151,6 +158,19 @@ private void registerFunctions(BeamSqlEnvBuilder sqlEnvBuilder) {
     udafDefinitions().forEach(udaf -> sqlEnvBuilder.addUdaf(udaf.udafName(), udaf.combineFn()));
   }
 
+  private static SqlTransform query(
+      String queryString, Object queryParameters, QueryParameterMode queryParameterMode) {
+    return builder()
+        .setQueryString(queryString)
+        .setQueryParameters(queryParameters)
+        .setQueryParameterMode(queryParameterMode)
+        .setUdafDefinitions(Collections.emptyList())
```

Comment:
This is a bit strange to me. Isn't the point of a builder that you only specify some of the args? I recognize that the oddness was pre-existing in this code.

Reply:
It looks like it was set up this way because some of the public methods (e.g. registerUdaf) themselves include some boilerplate building, which is hidden from the user.